src/android/HockeyApp.java

```diff
@@ -27,14 +31,70 @@
 import java.lang.Thread;
 import java.text.SimpleDateFormat;
 import java.util.Calendar;
+import java.util.ArrayList;
+
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.view.View;
 
 public class HockeyApp extends CordovaPlugin {
 
+    public static final long XWALK_SCREENSHOT_WAIT_MS = 5000;
+    public static final String XWALK_SCREENSHOT_CAPTURE_MSG = "captureXWalkBitmap";
+    public static final String XWALK_SCREENSHOT_BITMAP_MSG = "onGotXWalkBitmap";
+    
     public static boolean initialized = false;
     public static String appId;
+    public static Object monitor = new Object();
+    public static volatile Bitmap bitmap;
     
     private ConfiguredCrashManagerListener crashListener;
+    
+    // Integration with Crosswalk requires:
+    // - Crosswalk engine version 18 or later
+    // - Latest cordova-plugin-crosswalk-webview plugin     
+    private Bitmap getBitmap() {
+        bitmap = null;
+        boolean isCrosswalk = false;
+        try {
+            Class.forName("org.crosswalk.engine.XWalkWebViewEngine");
```

Comment:
Could you simply check the class name of the current webView property, instead of doing this? They would both have the same effect, but the intent would seemingly read a little cleaner if you did "webView.getClass().getName()" and compared that with the known type name. I think we care that the current WebView is in fact CrossWalk, not that the CrossWalk engine class is available in the app.